account_credit_control/models/credit_control_line.py

```diff
@@ -167,28 +167,41 @@ def _compute_partner_user_id(self):
 
     @api.model
     def _prepare_from_move_line(self, move_line, level, controlling_date,
-                                open_amount):
+                                open_amount, default_lines_vals):
         """ Create credit control line """
-        data = {
-            'date': controlling_date,
-            'date_due': move_line.date_maturity,
-            'state': 'draft',
-            'channel': level.channel,
-            'invoice_id': (move_line.invoice_id.id if
-                           move_line.invoice_id else False),
-            'partner_id': move_line.partner_id.id,
-            'amount_due': (move_line.amount_currency or move_line.debit or
-                           move_line.credit),
-            'balance_due': open_amount,
-            'policy_level_id': level.id,
-            'move_line_id': move_line.id,
-            'manual_followup': move_line.partner_id.manual_followup,
-        }
+        data = default_lines_vals.copy()
+        data.update(
+            {
+                'date': controlling_date,
+                'date_due': move_line.date_maturity,
+                'state': 'draft',
+                'channel': level.channel,
+                'invoice_id': (
+                    move_line.invoice_id.id if move_line.invoice_id else False
+                ),
+                'partner_id': move_line.partner_id.id,
+                'amount_due': (
+                    move_line.amount_currency
+                    or move_line.debit
+                    or move_line.credit
+                ),
+                'balance_due': open_amount,
+                'policy_level_id': level.id,
+                'move_line_id': move_line.id,
+                'manual_followup': move_line.partner_id.manual_followup,
+            }
+        )
         return data
 
     @api.model
-    def create_or_update_from_mv_lines(self, lines, level, controlling_date,
-                                       check_tolerance=True):
+    def create_or_update_from_mv_lines(
+        self,
+        lines,
+        level,
+        controlling_date,
+        default_lines_vals,
+        check_tolerance=True,
```

Comment:
add `default_lines_vals=None` as last argument for backward compatibility

Reply:
@qgroulard the default_lines_vals must be added *after* check_tolerance, in case someone was calling the method without keyword arguments.